test/dynamax.c

```diff
@@ -0,0 +1,1386 @@
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_HEAVY_SLAM].effect == EFFECT_HEAT_CRASH);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE, dynamax: TRUE); MOVE(opponent, MOVE_HEAVY_SLAM); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Max Strike!");
+        MESSAGE("Foe Wobbuffet used Heavy Slam!");
+        MESSAGE("The move was blocked by the power of Dynamax!");
+        NONE_OF { HP_BAR(player); }
+    }
+}
+
+SINGLE_BATTLE_TEST("(DYNAMAX) Dynamaxed Pokemon cannot be hit by OHKO moves")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_FISSURE].effect == EFFECT_OHKO);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_MACHAMP) { Ability(ABILITY_NO_GUARD); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE, dynamax: TRUE); MOVE(opponent, MOVE_FISSURE); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Max Strike!");
+        MESSAGE("Foe Machamp used Fissure!");
+        MESSAGE("Wobbuffet is unaffected!");
+        NONE_OF { HP_BAR(player); }
+    }
+}
+
+// can't be used at all in Raid, see "Documenting Dynamax"
+SINGLE_BATTLE_TEST("(DYNAMAX) Dynamaxed Pokemon are not affected by Destiny Bond")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Speed(50); };
+        OPPONENT(SPECIES_WOBBUFFET) { HP(1); Speed(100); }
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_DESTINY_BOND); MOVE(player, MOVE_TACKLE, dynamax: TRUE); }
+    } SCENE {
+        MESSAGE("Foe Wobbuffet used Destiny Bond!");
+        MESSAGE("Wobbuffet used Max Strike!");
+        MESSAGE("Foe Wobbuffet fainted!");
+        NONE_OF { HP_BAR(player); }
+    }
+}
+
+SINGLE_BATTLE_TEST("(DYNAMAX) Dynamaxed Pokemon are affected by Grudge")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Speed(50); };
+        OPPONENT(SPECIES_WOBBUFFET) { HP(1); Speed(100); }
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_GRUDGE); MOVE(player, MOVE_TACKLE, dynamax: TRUE); }
+    } SCENE {
+        MESSAGE("Foe Wobbuffet used Grudge!");
+        MESSAGE("Wobbuffet used Max Strike!");
+        MESSAGE("Wobbuffet's Tackle lost all its PP due to the GRUDGE!");
+        MESSAGE("Foe Wobbuffet fainted!");
+    }
+}
+
+SINGLE_BATTLE_TEST("(DYNAMAX) Dynamaxed Pokemon are not affected by phazing moves, but still take damage")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_DRAGON_TAIL].effect == EFFECT_HIT_SWITCH_TARGET);
+        ASSUME(gBattleMoves[MOVE_WHIRLWIND].effect == EFFECT_ROAR);
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_DRAGON_TAIL); MOVE(player, MOVE_TACKLE, dynamax: TRUE); }
+        TURN { MOVE(opponent, MOVE_WHIRLWIND); MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Max Strike!");
+        MESSAGE("Foe Wobbuffet used Dragon Tail!");
+        HP_BAR(player);
```

Comment:
I think this will fail here. if I understand correctly, `HP_BAR(player)` fails if the hp bar of the player changes at all. 
If dragon tail is supposed to do damage, here this will fail. maybe @mrgriffin can confirm. 
if thats the case the other uses of `HP_BAR(player)` should be looked at aswell

Reply:
I thought `HP_BAR` was supposed to confirm that there was a change in health. That's how I've been treating it.

Reply:
Your understanding is correct, I made a mistake when writing the documentation for `HP_BAR`, which has since been corrected :)